lib/1.4/dml-builtins.dml

```diff
@@ -539,6 +539,20 @@ template device {
     param objtype = "device";
 
     param simics_api_version auto;
+    param _api_5_or_older = simics_api_version == "5"
+        || simics_api_version == "4.8";
+    param _api_6_or_older = simics_api_version == "6" || _api_5_or_older;
+
+    param _DEPRECATE_PORT_PROXY_IFACES auto;
+    #if (!_DEPRECATE_PORT_PROXY_IFACES && !_api_6_or_older) {
+        error "The PORT_PROXY_IFACES feature can only be enabled"
+            + " with API=6 or older";
+    }
```

Comment:
If you remember my `--compat` rant, I think we one day may trip into the situation where we will need to offer an option to revert a particular deprecation/semantic change. But those will be very particular situations, and the deprecation of proxy ports and interfaces certainly isn't one of them.
My opinion is that this only makes sense if we eventually plan to introduce a _general_ `--compat` flag (as opposed to ad-hoc flags where we are selective about what kind of changes that users can turn off) -- I'm not particularly fond of a general `--compat` option, and I'm guessing you are even less so.
If we won't have a general `--compat` flag), there's no point to this. Why bother with executable documentation if you can just put a comment (here, or elsewhere?)

Reply:
We do have a generic compat flag, namely `--simics-api`.

Reply:
... where the meaning of `--simics-api=7` is sealed the day 7 is released.

Reply:
By "general" in this context I mean "granular and at will", e.g. `--compat=PORT_PROXY_IFACES`, like the opposite of `--deprecate`/`--no-compat`.

Reply:
And by "ad-hoc" I mean if we ever make a change that we discover "oops, people have legit reasons to want to reverse this change and keep using the modern API, but we still want its reversal to be 'opt-in' and lock out your model from certain features if necessary as to not shackle future development of DMLC" then instead of adding a full-fledged `--compat` we instead let people say `--enable-deprecated-[feature name]`

Reply:
Right, we can later choose to allow opting out from certain deprecations in API versions where they are enabled by default. If the need should arise.

Reply:
Indeed; and the gist of my argument is that seeing how we're very unlikely to want `--compat` in favor of ad-hoc compatibility flags, I don't think we want this executable documentaiton, especially if it incurs an unneeded auto param (unless we _do_ want people to access it, but then it shouldn't be `_` prefixed`.)

Reply:
I think the auto params are nice to have; in particular, good for workarounds in common code. The reason for the `_` prefix is that they will be removed when they are obsolete.